test/common/http/header_utility_test.cc

```diff
@@ -114,6 +114,35 @@ TEST_F(HeaderUtilityTest, RemovePortsFromHostConnectLegacy) {
   }
 }
 
+// Host's trailing dot from host header get removed
+TEST_F(HeaderUtilityTest, RemoveTrailingDotFromHost) {
+  const std::vector<std::pair<std::string, std::string>> host_headers{
+      {"localhost", "localhost"},          // w/o dot
+      {"localhost.", "localhost"},         // name w/ dot
+      {"", ""},                            // empty
+      {"192.168.1.1", "192.168.1.1"},      // ipv4
+      {"abc.com", "abc.com"},              // dns w/o dot
+      {"abc.com.", "abc.com"},             // dns w/ dot
+      {"abc.com:443", "abc.com:443"},      // dns port w/o dot
+      {"abc.com.:443", "abc.com:443"},     // dns port w/ dot
+      {"[fc00::1].:443", "[fc00::1]:443"}, // ipv6 w/ dot
```

Comment:
qq is this a valid input? Not sure when we'd get a . after an IP?

Reply:
Yah, its not valid input, its added by mistake and will remove it.